From 4c50db2fec23afdbcb12264ae01510553546a531 Mon Sep 17 00:00:00 2001
From: Cameron Clark <cameron.clark@hey.com>
Date: Fri, 14 Nov 2025 11:09:52 +0000
Subject: [PATCH 1/6] Adapt project service for single run mode

---
 tsc/internal/project/project.go | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tsc/internal/project/project.go b/tsc/internal/project/project.go
index 3a0cd8bc962..bd15137b684 100644
--- a/tsc/internal/project/project.go
+++ b/tsc/internal/project/project.go
@@ -3,6 +3,7 @@ package project
 import (
 	"context"
 	"fmt"
+	"runtime"
 	"slices"
 	"strconv"
 	"strings"
@@ -502,7 +503,9 @@ func (p *Project) CreateProgram() CreateProgramResult {
 	var newProgram *compiler.Program
 
 	createCheckerPool := func(program *compiler.Program) compiler.CheckerPool {
-		return newCheckerPool(p.host.sessionOptions.CheckerPoolOptions, program, p.log)
+		opts := p.host.sessionOptions.CheckerPoolOptions
+		opts.MaxCheckers = runtime.GOMAXPROCS(0)
+		return newCheckerPool(opts, program, p.log)
 	}
 	var cleanupModuleResolver func()
 	createModuleResolver := func(options module.ResolverOptions) module.Resolver {
